pyvista/utilities/reader.py

```diff
@@ -1405,8 +1457,8 @@ def active_datasets(self):
 
     def _set_filename(self, filename):
         """Set filename and update reader."""
-        self.filename = filename
-        self._directory = os.path.dirname(filename)
+        self._filename = filename
+        self._directory = os.path.join(os.path.dirname(filename))
```

Comment:
In this case `self._directory` means something different than for `DICOMReader`.  I'm not sure if we should rename to something else to prevent future refactoring of `self._directory` from messing up `PVDReader`.  I'm okay with the current state though as there is no clash now.

Reply:
The PVD Reader only uses its `._directory` attribute a single time, in `set_active_time_value`. It doesn't even have a property designation with a getter. I'm not 100% sure this won't break anything (would have to spend more time looking at the logic) but probably you could just get rid of it and replace the occurrence in that subclass as such:

Original:
```
    def set_active_time_value(self, time_value):  # noqa: D102
        self._active_datasets = [dataset for dataset in self.datasets if dataset.time == time_value]
        self._active_readers = [
            get_reader(os.path.join(self._directory, dataset.filename))
            for dataset in self.active_datasets
        ]
```

Becomes:
```
    def set_active_time_value(self, time_value):  # noqa: D102
        self._active_datasets = [dataset for dataset in self.datasets if dataset.time == time_value]
        self._active_readers = [
            get_reader(os.path.join(os.path.dirname(self.filename), dataset.filename))
            for dataset in self.active_datasets
        ]
```

Then there's no collision. Of course we ultimately don't even want `filename` and `directory` properties anyways and should change them to a unified `path` property in the near future.

Reply:
> In this case `self._directory` means something different than for `DICOMReader`. I'm not sure if we should rename to something else to prevent future refactoring of `self._directory` from messing up `PVDReader`. I'm okay with the current state though as there is no clash now.

Make it a double dunder. The name mangling will make a class local reference, allowing subclasses to override it without breaking any code:

```python
self.__directory = self._directory
```

Reply:
For reference, check out

[[Hettinger, Raymond. "Python's Class Development Toolkit". PyCon US 2013](https://www.youtube.com/watch?v=HTLu2DFOdTg)](https://www.youtube.com/watch?v=HTLu2DFOdTg)

starting at 33:25. I heart Raymond Hettinger.

Reply:
Good point @adam-grant-hendry.

Reply:
Resolved in 2c01008c8b034adc97d87c111fb7a39d4fdb71cd